public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -242,27 +242,22 @@ function getNewDashboardModelData(urlFolderId?: string): any {
   return data;
 }
 
-function updateQueriesWhenComingFromExplore(
+async function updateQueriesWhenComingFromExplore(
   dispatch: ThunkDispatch,
   dashboard: DashboardModel,
-  left: ExploreItemState
+  originPanelId: number,
+  queries: DataQuery[]
 ) {
-  // When returning to the origin panel from explore, if we're doing
-  // so with changes all the explore state is reset _except_ the queries
-  // and the origin panel ID.
-  const panelArrId = dashboard.panels.findIndex(panel => panel.id === left.originPanelId);
+  const panelArrId = dashboard.panels.findIndex(panel => panel.id === originPanelId);
 
   if (panelArrId > -1) {
-    dashboard.panels[panelArrId].targets = left.queries.map((query: DataQuery & { context?: string }) => {
+    dashboard.panels[panelArrId].targets = queries.map((query: DataQuery & { context?: string }) => {
       delete query.context;
       delete query.key;
```

Comment:
Since this is a explore specific, was thinking this "query cleanup" could be done before when dispatching  the setDashboardQueriesToUpdateAction action

Reply:
Good idea, will update